target/ppc/translate/fixedpoint-impl.c.inc

```diff
@@ -70,33 +70,51 @@ static bool do_ldst_X(DisasContext *ctx, arg_X *a, bool update,
     return do_ldst(ctx, a->rt, a->ra, cpu_gpr[a->rb], update, store, mop);
 }
 
-static bool do_ldst_quad(DisasContext *ctx, int ra, TCGv tgt_a, TCGv tgt_b,
-                         TCGv displ, bool store)
+static bool do_ldst_quad(DisasContext *ctx, arg_D *a, bool store, bool prefixed)
 {
-    TCGv ea;
+#if defined(TARGET_PPC64)
+    TCGv ea, tgt_a, tgt_b;
+    MemOp mop = MO_Q ^ ctx->default_tcg_memop_mask;
+
+    if (!store && a->ra == a->rt) {
+        gen_inval_exception(ctx, POWERPC_EXCP_INVAL_INVAL);
+        return true;
+    }
+
+    if (!prefixed && !(ctx->insns_flags2 & PPC2_LSQ_ISA207)) {
+        if (ctx->pr) {
+            /* lq and stq were privileged prior to V. 2.07 */
+            gen_priv_exception(ctx, POWERPC_EXCP_PRIV_OPC);
+            return true;
+        }
+
+        if (ctx->le_mode) {
+            gen_align_no_le(ctx);
+            return true;
+        }
+    }
 
     gen_set_access_type(ctx, ACCESS_INT);
     ea = tcg_temp_new();
-    do_ea_calc(ctx, ra, displ, ea);
+    do_ea_calc(ctx, a->ra, tcg_constant_tl(a->si), ea);
+
+    if (prefixed || !ctx->le_mode) {
+        tgt_a = cpu_gpr[a->rt];
+        tgt_b = cpu_gpr[a->rt + 1];
+    } else {
+        tgt_a = cpu_gpr[a->rt + 1];
+        tgt_b = cpu_gpr[a->rt];
+    }
 
     if (tb_cflags(ctx->base.tb) & CF_PARALLEL) {
         if (HAVE_ATOMIC128) {
             TCGv_i32 oi = tcg_temp_new_i32();
             tcg_gen_movi_i32(oi, make_memop_idx(MO_Q, ctx->mem_idx));
```

Comment:
The endianness in `opidx` seems to be ignored, since the `cpu_atomic_*` method carries the endianess in the name. Testing both options led to the same result. But we already have a `mop` with the correct value, so I guess it doesn't hurt.

Reply:
Fixes in https://github.com/PPC64/qemu/pull/14/commits/c6d056046c12aba03406d0e28590622b116c71e1